kolibri/core/deviceadmin/utils.py

```diff
@@ -205,6 +206,16 @@ def perform_vacuum(database=db.DEFAULT_DB_ALIAS):
             logger.error(new_msg)
         else:
             logger.info("Sqlite database Vacuum finished.")
+    elif connection.vendor == "postgresql":
+        morango_models = [
+            m
+            for m in apps.get_models(include_auto_created=True)
+            if "morango.models" in str(m)
+        ]
```

Comment:
Is the time it takes to vacuum a table proportional to the size of the table, or to how much junk space there is "to vacuum"?

The only Morango tables with much churn are the [`Buffer`](https://github.com/learningequality/morango/blob/master/morango/models/core.py#L412) and the [`RecordMaxCounterBuffer`](https://github.com/learningequality/morango/blob/master/morango/models/core.py#L520).

While it shouldn't hurt to vacuum the others as well, some of them (e.g. the `Store`) can be very large, but under normal circumstances not have any churn at all (they're added and updated, but never deleted), and since we're running this vacuuming automatically, we'll want to keep it as efficient as we can.

Reply:
The time it will take will depend on how disorganized the tables are. Usually large tables are more disorganized than small ones but the size of the table is not an accurate rule of thumb. Such as this is, the size reduction won't be as big as executing a _full vacuum_, but it will reorganize the data to be performant. In any case with this implementation the tables are not locked while vacuum and the application will start normally. The user should not notice anything.

Without adding _full_ to the vacuum command this is a non-blocking command. I'm thinking of adding a `--full `option (to be false by default) to the command so in case we want to execute it in a cron, it would be a complete vacuum of the tables. That would produce table locks and will reduce much more the size of the tables, but it would not be executed when kolibri starts, only on demand by a command. 

What do you think? having a possiblity to run it under cron for a full vacuum would be useful for us? In that case, limiting it to the two tables you mention makes more sense as they won't be available while vacuum is executed.

Reply:
Have discussed this with @jredrejo and it seems like his plan is good!

1) The non-full will run on all tables because it is non-locking
2) Will add a 'full' flag that will run only on those two tables because that will reclaim disk space for tables that grow and then shrink again.

Reply:
That sounds good, yeah -- for Cloud Kolibri and KDP we'll want to have that cron job running regularly to keep things in check (as there's a ton of churn on those tables).

Reply:
Done